File: 프로그래머스/0/120840. 구슬을 나누는 경우의 수/구슬을 나누는 경우의 수.cpp
#include <bits/stdc++.h>

using namespace std;

int combi(int start, vector<int> &v, int n, int r) {
    if(v.size() == r) {
        return 1;
    }
    int cnt = 0;

    for(int i = start + 1; i < n; i++) {
        v.push_back(i);
        cnt += combi(i, v, n, r);
        v.pop_back();
    }
    return cnt;
}

int solution(int balls, int share) {
    vector<int> v;
    return combi(-1, v, balls, share);
}

File: 프로그래머스/0/120840. 구슬을 나누는 경우의 수/구슬을 나누는 경우의 수.swift
import Foundation

func solution(_ balls:Int, _ share:Int) -> Int {
    let min = balls - share < share ? balls - share : share
    var result: Double = 1
    
    for i in 0..<min {
        result *= Double(balls - i) / Double(i + 1)
    }
    
    return Int(round(result))
}